src/wp-includes/style-engine/class-wp-style-engine.php

```diff
@@ -0,0 +1,560 @@
+	}
+
+	/**
+	 * Returns a store by store key.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param string $store_name A store key.
+	 *
+	 * @return WP_Style_Engine_CSS_Rules_Store
+	 */
+	public static function get_store( $store_name ) {
+		return WP_Style_Engine_CSS_Rules_Store::get_store( $store_name );
+	}
+
+	/**
+	 * Returns classnames and CSS based on the values in a styles object.
+	 * Return values are parsed based on the instructions in BLOCK_STYLE_DEFINITIONS_METADATA.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $block_styles The style object.
+	 * @param array $options      {
+	 *     Optional. An array of options. Default empty array.
+	 *
+	 *     @type bool        $convert_vars_to_classnames Whether to skip converting incoming CSS var patterns, e.g., `var:preset|<PRESET_TYPE>|<PRESET_SLUG>`, to var( --wp--preset--* ) values. Default `false`.
+	 *     @type string      $selector                   Optional. When a selector is passed, the value of `$css` in the return value will comprise a full CSS rule `$selector { ...$css_declarations }`,
+	 *                                                   otherwise, the value will be a concatenated string of CSS declarations.
+	 * }
+	 *
+	 * @return array {
+	 *     @type string                $classnames   Classnames separated by a space.
+	 *     @type array<string, string> $declarations An array of property/value pairs representing parsed CSS declarations.
+	 * }
+	 */
+	public static function parse_block_styles( $block_styles, $options ) {
+		$parsed_styles = array(
+			'classnames'   => array(),
+			'declarations' => array(),
+		);
+		if ( empty( $block_styles ) || ! is_array( $block_styles ) ) {
+			return $parsed_styles;
+		}
+
+		// Collect CSS and classnames.
+		foreach ( static::BLOCK_STYLE_DEFINITIONS_METADATA as $definition_group_key => $definition_group_style ) {
+			if ( empty( $block_styles[ $definition_group_key ] ) ) {
+				continue;
+			}
+			foreach ( $definition_group_style as $style_definition ) {
+				$style_value = _wp_array_get( $block_styles, $style_definition['path'], null );
+
+				if ( ! static::is_valid_style_value( $style_value ) ) {
+					continue;
+				}
+
+				$parsed_styles['classnames']   = array_merge( $parsed_styles['classnames'], static::get_classnames( $style_value, $style_definition ) );
+				$parsed_styles['declarations'] = array_merge( $parsed_styles['declarations'], static::get_css_declarations( $style_value, $style_definition, $options ) );
+			}
+		}
+
+		return $parsed_styles;
+	}
+
+	/**
+	 * Returns classnames, and generates classname(s) from a CSS preset property pattern, e.g., '`var:preset|<PRESET_TYPE>|<PRESET_SLUG>`'.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $style_value      A single raw style value or css preset property from the $block_styles array.
+	 * @param array $style_definition A single style definition from BLOCK_STYLE_DEFINITIONS_METADATA.
+	 *
+	 * @return array[] An array of CSS classnames.
```

Comment:
```suggestion
	 * @return string[] An array of CSS classnames.
```

Reply:
That was dumb. Thanks for catching that. Clearly I need to do more Typescript. :trollface: